src/Std/Time.lean

```diff
@@ -0,0 +1,173 @@
+
+- **`LocalTime`**: Represents a time of day in the format `HH:mm:ss.SSSSSSSSS`.
+
+## Date and time
+Combines date and time into a single representation, useful for precise timestamps and scheduling.
+
+- **`LocalDateTime`**: Represents both date and time in the format `YYYY-MM-DDTHH:mm:ss.SSSSSSSSS`.
+- **`Timestamp`**: Represents a point in time with second-level precision. It starts on the UNIX
+epoch and it should be used when you receive or need to send timestamps to another systems.
+
+## Zoned date and times.
+Combines date, time and time zones.
+
+- **`DateTime`**: Represents both date and time but with a time zone in the type constructor.
+- **`ZonedDateTime`**: An existential version of the `DateTime`.
+
+## Duration
+Represents spans of time and the difference between two points in time.
+
+- **`Duration`**: Represents the time span or difference between two `Timestamp`s values with nanosecond precision.
+
+# Formats
+
+Format strings are used to convert between `String` representations and date/time types, e.g, `YYYY-MM-DD'T'hh:mm:ss.sssZ`.
+The supported formats include:
+
+- **Year:**
+  - `YYYY`: Four-digit year (e.g., 2024)
+  - `YY`: Two-digit year (e.g., 24 for 2024)
+- **Month:**
+  - `MMMM`: Full month name (e.g., January, February)
+  - `MMM`: Abbreviated month name (e.g., Jan, Feb)
+  - `MM`: Two-digit month (e.g., 01 for January)
+  - `M`: One or two-digit month (e.g., 1 for January, 10 for October)
+- **Day:**
+  - `DD`: Two-digit day of the month (e.g., 01, 02)
+  - `D`: One or two-digit day of the month (e.g., 1, 2)
+  - `d`: One or two-digit day of the month with space padding (e.g., " 1", "12")
+- **Day of Week:**
+  - `EEEE`: Full name of the day of the week (e.g., Monday, Tuesday)
+  - `EEE`: Abbreviated day of the week (e.g., Mon, Tue)
+- **Hour:**
+  - `hh`: Two-digit hour in 24-hour format (e.g., 13, 14)
+  - `h`: One or two-digit hour in 24-hour format (e.g., 1, 2)
+  - `HH`: Two-digit hour in 12-hour format (e.g., 01, 02)
+  - `H`: One or two-digit hour in 12-hour format (e.g., 1, 2)
+- **AM/PM Indicator:**
+  - `AA`: Uppercase AM/PM (e.g., AM, PM)
+  - `aa`: Lowercase am/pm (e.g., am, pm)
+- **Minute:**
+  - `mm`: Two-digit minute (e.g., 01, 02)
+  - `m`: One or two-digit minute (e.g., 1, 2)
+- **Second:**
+  - `sss`: Three-digit milliseconds (e.g., 001, 202)
+  - `ss`: Two-digit second (e.g., 01, 02)
+  - `s`: One or two-digit second (e.g., 1, 2)
+- **Timezone:**
+  - `ZZZZZ`: Full timezone offset with hours and minutes (e.g., +03:00)
+  - `ZZZZ`: Timezone offset without the colon (e.g., +0300)
+  - `ZZZ`: Like `ZZZZ`, but with "UTC" for UTC
+  - `Z`: Like `ZZZZZ`, but with "Z" for UTC
+  - `z`: Timezone name (e.g., Brasilia Standard Time)
+
+# Macros
+
+In order to help the user build dates easily, there are a lot of macros available for creating dates.
+
+- **`date% DD-MM-YYYY`**: Defines a date in the `DD-MM-YYYY` format.
```

Comment:
I remember at one poing being told that the naming convention of `foo%` elaborators was to indicate implementation details; has this stance changed?

Another option here would be `date(DD-MM-YYYY)`, which maybe avoids confusion  around precedence of subtraction

Reply:
It seems that this discussion started on a Zulip thread that ended with no conclusion. As other elaborators have folllowed a similar pattern, this one will follow this pattern too. The other option avoids some problems but it will make `date` a keyword that is not reasonable because people will probably use `date` as a variable.

Reply:
I agree that the conclusion was unclear on the zulip thread.

However, `date(` would not make `date` a keyword, nor would it prevent a local variable being called `date`:` `f(x)` is not legal function application syntax in Lean, so it is fine to create new "keywords" that end in `(` without affecting function application.

Reply:
I'm planning to discuss this with the team at Lean FRO to help in the standardization of term elaborators in the Std library.

Reply:
@eric-wieser The Std team decided to follow the format you suggested :)